src/products/services/products/products.service.spec.ts

```diff
@@ -154,12 +155,24 @@ describe('ProductsService', () => {
         .mockReturnValue({
           exec: jest.fn().mockResolvedValueOnce(deletedProduct)
         } as any);
-      const result = await service.delete(mockProduct.id);
-      expect(model.findByIdAndUpdate).toHaveBeenCalledWith(
-        mockProduct.id,
-        { isHidden: true }
-      );
-      expect(result).toEqual(deletedProduct);
+      mockProduct = await service.delete(mockProduct.id);
+      expect(mockProduct).toEqual(deletedProduct);
```

Comment:
Here we still want to compare the `result` to the `deletedProduct`, so it should still be `expect(result).toEqual(deletedProduct);` to use the `result` from the service.